libs/dal/src/repositories/integration/integration.entity.ts

```diff
@@ -0,0 +1,20 @@
+import { ProviderEnum } from '@notifire/shared';
+
+interface ICredentials {
+  apiKey: string;
+  secretKey: string;
+}
+
+export class IntegrationEntity {
+  _id?: string;
+
+  _applicationId: string;
+
+  _organizationId: string;
+
+  providerId: string;
+
+  channel: ProviderEnum;
+
+  credentials: ICredentials[];
```

Comment:
```suggestion
  credentials: ICredentials;
```

Would change this to an object instead of array